src/libp2p/service.rs

```diff
@@ -674,54 +678,89 @@ async fn handle_chain_exchange_event<DB>(
     ce_event: request_response::Event<ChainExchangeRequest, ChainExchangeResponse>,
```

Comment:
The changes to this function are for fetching block headers when in `stateless` mode? If so, let's reconsider. For keeping the N most recent block headers, we can just get them from the gossip events.

Reply:
It's for `If a block is requested that we do not have, forward the request to other peers and cache the result.`

> If we do not respond to ChainRequests, other peers will eventually disconnect from us. Not a show-stopper, but we'd like for them to stay connected to us for as long as possible.

Reply:
Remove the logic that forwards chain exchange requests, as discussed